accounts/service.go

```diff
@@ -900,23 +845,24 @@ func (s *InterceptorService) removePayment(ctx context.Context,
 		return nil
 	}
 
-	account, err := s.store.Account(ctx, pendingPayment.accountID)
-	if err != nil {
-		return err
-	}
-
 	pendingPayment.cancel()
 	delete(s.pendingPayments, hash)
```

Comment:
Just noting that similar to the other PR, this will now run before we've checked if there's an account id for the pendingPayment. I do that should be unreachable though, and probably actually correct behaviour, so don't really think this needs to change.

Reply:
can also move it below the db call yeah?

Reply:
Hmmm yes, specifically since these two marked functions cannot error, and we don't shutdown the `accounts` service if the public `RemovePayment` function errors, I think we should move this to after the db call.

Otherwise we can get a diff between the in-memory layer of the `service` and the `db` layer. For the other call sites that calls `removePayment`, it shouldn't matter as they'll shutdown the service, and it'll be resolved on the next startup.

Reply:
updated 👍